eva-accession-unmapped-variants-pipeline/generate-unmapped-variants-report.py

```diff
@@ -0,0 +1,180 @@
+            select 
+                '{database_name}' as database_name,
+                lnk.subsnp_id, lnk.snp_id, 
+                seq3.line as seq3_line, seq3.type as seq3_type, 
+                seq5.line as seq5_line, seq5.type as seq5_type, 
+                lnk.create_time, lnk.last_updated_time
+            from
+                dbsnp_{database_name}.snpsubsnplink as lnk
+                left join dbsnp_{database_name}.contigloc_snp_ids_all_builds all_snps on lnk.snp_id = all_snps.snp_id
+                join dbsnp_{database_name}.subsnpseq3 as seq3 on lnk.subsnp_id = seq3.subsnp_id
+                join dbsnp_{database_name}.subsnpseq5 as seq5 on lnk.subsnp_id = seq5.subsnp_id
+            where all_snps.snp_id is null
+        );
+        """.format(**species_info)
+    return (os.linesep*2).join([work_mem_clause, temp_table_creation_query, subsnpseq_index_query,
+                                textwrap.dedent(contigloc_snp_id_query + unmapped_variants_report_query)])
+
+
+def write_report_query_file(species_info):
+    species_info["report_query_file"] = os.path.sep.join([species_info["report_folder"],
+                                                          "unmapped_variants_query.sql"])
+    with open(species_info["report_query_file"], "w") as report_query_file_handle:
+        report_query_file_handle.write(get_report_creation_query(species_info).lstrip())
+
+
+def run_report_for_species(species_info):
+    if "contigloc_table_list" in species_info:
+        os.makedirs(species_info["report_folder"], exist_ok=True)
+        logger.info("Creating report for species: {database_name} in the folder {report_folder}"
+                    .format(**species_info))
+        write_report_query_file(species_info)
+
+        dependency_argument_for_bsub = ""
+        # Only run one job at a time in a given Postgres host machine
+        if species_info["pg_host"] in prev_job_in_host:
+            dependency_argument_for_bsub = '-w "ended({0})" '.format(prev_job_in_host[species_info["pg_host"]])
+        prev_job_in_host[species_info["pg_host"]] = species_info["bsub_job_name"]
+
+        bsub_command_to_run = ("cd {report_folder} && bsub " + dependency_argument_for_bsub
+                               + "-J {bsub_job_name} "
+                                 "-o {bsub_job_name}.log "
+                                 "-e {bsub_job_name}.err "
+                               + "psql -U dbsnp -h {pg_host} -p {pg_port} -d dbsnp_{dbsnp_build} "
+                                 "-f {report_query_file} -v ON_ERROR_STOP=1").format(**species_info)
+
+        with open(final_command_file, "a") as command_file_handle:
+            command_file_handle.write(bsub_command_to_run + os.linesep)
+
+
+def check_prerequisites_for_species(species_info):
+    if not does_subsnpseq_table_exist_for_species(species_info):
+        logger.error("Species {0} does not have one or both of the tables: SUBSNPSEQ3 and SUBSNPSEQ5"
```

Comment:
Has a single table missing been observed in any database?

Reply:
I started off with the assumption that this would be missing for some. But I was pleasantly surprised to be wrong here!